src/Transport/GrpcTransport.php

```diff
@@ -283,4 +307,66 @@ private static function loadClientCertSource(callable $clientCertSource)
     {
         return call_user_func($clientCertSource);
     }
+
+    /**
+     * @param Call $call
+     * @param array $headers
+     *
+     * @return void
+     */
+    private function logRequest(Call $call, array $headers)
+    {
+        $logger = $this->logger;
+        $timestamp = date(DATE_RFC3339);
+
+        $debugEvent = [
+            'timestamp' => $timestamp,
+            'severity' => 'DEBUG',
+            'jsonPayload' => [
+                'request.rpcName' => $call->getMethod(),
+                'request.headers' => $headers,
+                'request.payload' => $call->getMessage()
+            ]
+        ];
+
+        $logger->debug(json_encode($debugEvent));
+    }
+
+    /**
+     * @param mixed $response
+     * @param mixed $status
+     *
+     * @return void
+     */
+    private function logResponse(mixed $response, mixed $status)
+    {
+        $logger = $this->logger;
+        $timestamp = date(DATE_RFC3339);
+
+        // In the case we have a $status->code != Code::OK
+        // from the request, we log the status only and we do not have
+        // a response.
```

Comment:
I am not crazy about the fact that we log two events, the info AND the debug event, separately. Wouldn't it be better to just log one event, such as:

```php
            $debugEvent = [
                'timestamp' => $timestamp,
                'severity' => 'DEBUG',
                'jsonPayload' => [
                    'response.headers' => $response ? $status->metadata : null,
                    'response.payload' => $response ?? null,
                    'response.status' => $status->code
                ]
            ];
```

The way you have it, one is `INFO` and one is `DEBUG`, so we'd only get to use one, but I assume that's not an important part of the requirements.

Reply:
I get what you mean and I agree.

But the current requirements for the project states that `status` is an `info` level log:
![image](https://github.com/user-attachments/assets/717ec433-377c-4588-aaf2-5c7375c33d2d)

Reply:
This is maybe something you should confirm with the group then (or maybe you already have)... that we want a specific logging event just for `response.status`. If so, then LGTM

Reply:
one issue with having multiple events is when we have multiple requests showing up, we can't be sure of which `response.status` is connected to which `response.header` and `response.payload`. We need to either log these things in the same event, or have a trace ID which ties them together.

Reply:
There is a `traceId` field but it is meant for open telemetry. After yesterday's meeting we stablished that we can in fact switch logs levels to different levels.